org.bndtools.templating/src/org/bndtools/templating/PostProcessor.java

```diff
@@ -0,0 +1,43 @@
+package org.bndtools.templating;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.osgi.annotation.versioning.ProviderType;
+
+@ProviderType
+public interface PostProcessor {
+
+    /**
+     * <p>
+     * Service registration property indicating the kind(s) of templates this post-processor can target. Currently
+     * defined values are 'project', 'workspace' and 'bndrun'. The type of this property is String+. The post-processor
+     * shall be ignored if it does not specify at least one kind.
+     * </p>
+     */
+    public static String PROP_TEMPLATE_KIND = "kind";
+
+    /**
+     * Post-process the output of the template.
+     *
+     * @param templateKind
+     *            The kind of template that was used.
+     * @param template
+     *            A reference to the template; NB a post-processor must NOT invoke {@link Template#generateOutputs(Map)}
+     *            or {@link Template#generateOutputs(Map, IProgressMonitor)} at this point, but is permitted to call any
+     *            of the other methods of {@link Template}.
+     * @param templateParams
+     *            The input parameters that were supplied to the template.
+     * @param resources
+     *            The resources generated by the template. A post-processor may add, remove or modify any entries in
+     *            this map.
+     * @param monitor
+     *            A monitor for progress reporting.
+     * @return
+     * @throws Exception
+     */
+    IStatus modifyOutputs(String templateKind, Template template, Map<String,List<Object>> templateParams, ResourceMap resources, IProgressMonitor monitor) throws Exception;
```

Comment:
If at all possible I'd like this decoupled of Eclipse interfaces.
bndtools could provide internal conversion.

I gather that a plugin can modify the template and when I return bndtools will actually generate them on the filesystem?